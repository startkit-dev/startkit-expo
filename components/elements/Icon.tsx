import { clsx } from "clsx"
import { type LucideProps, X } from "lucide-react-native"

/**
 * List all icons used in the app here.
 *
 * This is used to limit the bundle size by only importing the icons we use.
 */
export const icons = {
  x: X
}

export type IconProps = LucideProps & {
  name: keyof typeof icons
}

/**
 * A dark-mode aware wrapper around the Lucide vector icon component which
 * applies our basic styles.
 *
 * If a color is provided, it will be used. Otherwise, the default color will be
 * used.
 *
 * For icons, see: https://lucide.dev
 */
export function Icon({ name, color, ...props }: IconProps) {
  const LucideIcon = icons[name]

  return (
    <LucideIcon
      className={clsx(!color && "text-black dark:text-white")}
      color={color}
      {...props}
    />
  )
}
