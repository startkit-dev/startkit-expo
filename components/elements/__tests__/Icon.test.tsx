import { create } from "react-test-renderer"

import { Icon } from "../Icon"

test("renders correctly", () => {
  const tree = create(<Icon name="x" />).toJSON()

  expect(tree).toMatchSnapshot()
})
