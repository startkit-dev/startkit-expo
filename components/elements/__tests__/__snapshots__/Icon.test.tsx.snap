// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="currentColor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
        "strokeLinecap",
        "strokeLinejoin",
      ]
    }
    stroke={
      {
        "type": 2,
      }
    }
    strokeLinecap={1}
    strokeLinejoin={1}
    strokeWidth="2"
  >
    <RNSVGLine
      x1="18"
      x2="6"
      y1="6"
      y2="18"
    />
    <RNSVGLine
      x1="6"
      x2="18"
      y1="6"
      y2="18"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
